rdrf/rdrf/views/form_view.py

```diff
@@ -1565,10 +1565,16 @@ class FileUploadView(View):
 
     @login_required_method
     def get(self, request, registry_code, file_id):
-        data, filename = filestorage.get_file(file_id)
-        if data is not None:
-            response = FileResponse(data, content_type='application/octet-stream')
-            response['Content-disposition'] = 'filename="%s"' % filename
+        file_info = filestorage.get_file(file_id)
+        if file_info.patient:
+            security_check_user_patient(request.user, file_info.patient)
+        elif file_info.uploaded_by:
+            if user_is_patient_type(request.user) and request.user != file_info.uploaded_by:
+                raise PermissionDenied
+
+        if file_info.item is not None:
+            response = FileResponse(file_info.item, content_type='application/octet-stream')
```

Comment:
This seems strange. Shouldn't we store the mime type on that file info object and set that on the response instead of handling all files a binary?
It is better to add another JIRA task for this though to finish this one asap.

Reply:
Yes. FileStorage should be modified to include mime-type.